### PR TITLE
Feature/multi llm

### PR DESCRIPTION
**Pull request by iorisa**
**Features**
- Allow the `Action` object to use the specified LLM configuration; 
- Support for multiple LLM configurations. If no LLM is specified, the default LLM configuration is used.

### CODE REVIEW

File: config/config2.example.yaml [hunk 1]
```diff
@@ -59,3 +59,21 @@ iflytek_api_key: "YOUR_API_KEY"
...
+models:
+#  "YOUR_MODEL_NAME_1": # model: "gpt-4-turbo"  # or gpt-3.5-turbo
+#    api_type: "openai"  # or azure / ollama / groq etc.
+#    base_url: "YOUR_BASE_URL"
+#    api_key: "YOUR_API_KEY"
+#    proxy: "YOUR_PROXY"  # for LLM API requests
+#    # timeout: 600 # Optional. If set to 0, default value is 300.
+#    # Details: https://azure.microsoft.com/en-us/pricing/details/cognitive-services/openai-service/
+#    pricing_plan: "" # Optional. Use for Azure LLM when its model name is not the same as OpenAI's
+#  "YOUR_API_TYPE": # api_type: "openai"  # or azure / ollama / groq etc.
```
Comment (better629) [inline config/config2.example.yaml:72]: YOUR_MODEL_NAME_2 or YOUR_API_TYPE

File: metagpt/configs/models_config.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        Returns:
+            Optional[LLMConfig]: LLMConfig object if found, otherwise None.
+        """
+        if not name_or_type:
+            return None
+        model = self.models.get(name_or_type)
+        if model:
+            return model
+        for m in self.models.values():
+            if m.api_type == name_or_type:
```
Comment (better629) [inline metagpt/configs/models_config.py:110]: If there are multi model_name1 model_name2 which both use api_type=openai. Here will return the first one if name_or_type="openai", whether it meets expectation?

Comment (iorisa) [reply metagpt/configs/models_config.py:110]:
```
models:
#  "YOUR_MODEL_NAME_1 or YOUR_API_TYPE_1": # model: "gpt-4-turbo"  # or gpt-3.5-turbo
#    api_type: "openai"  # or azure / ollama / groq etc.
#    base_url: "YOUR_BASE_URL"
#    api_key: "YOUR_API_KEY"
#    proxy: "YOUR_PROXY"  # for LLM API requests
#    # timeout: 600 # Optional. If set to 0, default value is 300.
#    # Details: https://azure.microsoft.com/en-us/pricing/details/cognitive-services/openai-service/
#    pricing_plan: "" # Optional. Use for Azure LLM when its model name is not the same as OpenAI's
#  "YOUR_MODEL_NAME_2 or YOUR_API_TYPE_2": # api_type: "openai"  # or azure / ollama / groq etc.
#    api_type: "openai"  # or azure / ollama / groq etc.
#    base_url: "YOUR_BASE_URL"
#    api_key: "YOUR_API_KEY"
#    proxy: "YOUR_PROXY"  # for LLM API requests
#    # timeout: 600 # Optional. If set to 0, default value is 300.
#    # Details: https://azure.microsoft.com/en-us/pricing/details/cognitive-services/openai-service/
#    pricing_plan: "" # Optional. Use for Azure LLM when its model name is not the same as OpenAI's
```
As expected.

The configuration is in the form of a map, so the user should be aware that keys must be unique. If fetching without specifying a key, it means any match will suffice.

File: config/config2.example.yaml (hunk 1 above)
Comment (iorisa) [reply config/config2.example.yaml]: fixed

**Comment by better629** [comment]
@iorisa see the pre-commit checks

**Comment by better629** [comment]
lgtm

**Review by geekan — APPROVED** [review]

**Pull request merged**